src/perfrecord.cpp

```diff
@@ -0,0 +1,93 @@
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QDebug>
+#include <QFileInfo>
+
+#include "perfrecord.h"
+
+PerfRecord::PerfRecord(QObject* parent)
+    : QObject(parent)
+    , m_PerfRecordProcess(new QProcess(this))
+    , m_perfCommand()
+    , m_outputPath()
+{
+}
+
+PerfRecord::~PerfRecord() = default;
+
+void PerfRecord::record(const QStringList &perfOptions, const QString &outputPath, const QString &exePath, const QStringList &exeOptions)
+{
+    QFileInfo info(exePath);
+    if (!info.exists()) {
+        emit recordingFailed(tr("File '%1' does not exist.").arg(exePath));
+        return;
+    }
+    if (!info.isFile()) {
+        emit recordingFailed(tr("'%1' is not a file.").arg(exePath));
+        return;
+    }
+    if (!info.isReadable()) {
+        emit recordingFailed(tr("File '%1' is not readable.").arg(exePath));
+        return;
+    }
+
+    connect(m_PerfRecordProcess, static_cast<void (QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
+            [this] (int exitCode, QProcess::ExitStatus exitStatus) {
+                qDebug() << exitCode << exitStatus;
+
+                if (exitCode == EXIT_SUCCESS && QFileInfo::exists(m_outputPath)) {
+                    emit recordingFinished(m_outputPath);
+                } else {
+                    emit recordingFailed(tr("Failed to record perf data, error code %1.").arg(exitCode));
+                }
+            });
+
+    connect(m_PerfRecordProcess, &QProcess::errorOccurred,
+            [this] (QProcess::ProcessError error) {
+                qWarning() << error << m_PerfRecordProcess->errorString();
+
+                emit recordingFailed(m_PerfRecordProcess->errorString());
+            });
+
+    m_outputPath = outputPath;
+
+    QLatin1String perfBinary("perf");
+    QStringList perfCommand;
+    perfCommand << QLatin1String("record") << QLatin1String("-o") << m_outputPath;
+    perfCommand.append(perfOptions);
+    perfCommand.append(exePath);
+    perfCommand.append(exeOptions);
+    m_perfCommand = perfBinary + QLatin1Char(' ') + perfCommand.join(QLatin1Char(' '));
+
+    m_PerfRecordProcess->start(perfBinary, perfCommand);
```

Comment:
lower-case the first p, then use `m_perfRecordProcess(QStringLiteral("perf"), m_perfCommand);`